Quaver.Shared/Database/Playlists/PlaylistManager.cs

```diff
@@ -513,6 +522,62 @@ public static void UpdateMapInPlaylists(Map outdated, Map newMap)
             });
         }
 
+        /// <summary>
+        ///     Imports an online playlist
+        /// </summary>
+        /// <param name="onlineId"></param>
+        public static void ImportPlaylist(int onlineId)
+        {
+            Logger.Important($"Importing playlist: ID {onlineId}", LogType.Runtime);
+            var playlistResponse = new APIRequestPlaylistInformation(onlineId).ExecuteRequest();
+            if (playlistResponse == null || playlistResponse.Status != 200)
+            {
+                Logger.Important(
+                    $"Failed retrieving playlist information with error code: {playlistResponse?.Status ?? -1}",
+                    LogType.Network);
+                NotificationManager.Show(NotificationLevel.Error, "Failed to retrieve playlist information");
+                return;
+            }
+
+            var playlistAlreadyExistsLocally = false;
+
+            var playlist = Playlists.Find(p => p.OnlineMapPoolId == onlineId);
+            if (playlist == null)
+            {
+                var newPlaylist = new Playlist()
+                {
+                    Name = playlistResponse.PlaylistInformation.Name,
+                    Description = playlistResponse.PlaylistInformation.Description,
+                    Creator = playlistResponse.PlaylistInformation.OwnerUsername,
+                    OnlineMapPoolId = playlistResponse.PlaylistInformation.Id,
+                    OnlineMapPoolCreatorId = playlistResponse.PlaylistInformation.OwnerId
+                };
+
+                AddPlaylist(newPlaylist);
+                playlist = Playlists.Find(p => p.OnlineMapPoolId == onlineId);
```

Comment:
Is it possible to just do playlist = new Playlist() above without this Find?

Reply:
@YaLTeR  Do you mean `playlist = newPlaylist`?

It's the same object, so not sure why the find is there.

Reply:
Yeah essentially, even without newPlaylist was my suggestion. Just assign to playlist right away.

Reply:
I didn't know the database insert also assigns the ID, learned something new